examples/Protobuf/messages/CMakeLists.txt

```diff
@@ -1,7 +1,41 @@
+# Example for how to compile .proto files into a library
+# Both the 'Module Compatible' and new way are shown
 
-# required since no linking
-include_directories(${PROTOBUF_INCLUDE_DIR})
-PROTOBUF_GENERATE_CPP(MESSAGES_SRC MESSAGES_HEADER messages.proto)
-add_library(messages ${MESSAGES_HEADER} ${MESSAGES_SRC})
+set(MESSAGES_PROTOS messages.proto)
+set(MESSAGES_SRCS)
+set(MESSAGES_HDRS)
 
+#Code Generation
+if(protobuf_MODULE_COMPATIBLE) #Legacy Support
```

Comment:
You can name example as you wish. It may be `protobuf-legacy` or `protobuf-compatible` or anything. The point is to have two examples instead of one hairy `if(..._COMPATIBLE)` version.